src/views/Exercises/Edit.vue

```
@@ -1,9 +1,39 @@
 <template>
   <div>
-    <RouterView />
+    <LoadingMessage
+      v-if="loaded === false"
+      ref="loadingMessageComponent"
+      :load-failed="loadFailed"
+    />
+    <div 
+      v-else
+    >
+      <RouterView />
+    </div>
   </div>
 </template>
 
 <script>
-export default {};
+import LoadingMessage from '@/components/LoadingMessage';
+export default {
+  components: {
+    LoadingMessage,
+  },
+  data() {
+    return {
+      loaded: false,
+      loadFailed: false,
+    };
+  },
+  mounted() {
+    const id = this.$route.params.id;
+    
+    this.$store.dispatch('exerciseDocument/bind', id)
+      .then(() => {
+        this.loaded = true;
+      }).catch(() => {
+        this.loadFailed = true;
```

Please also throw the error which is caught here (it'll help when we have Sentry integration)